### PR TITLE
qa: Fix service flag comparison check in rpc_net test

### PR DESCRIPTION
**Pull request by luke-jr**
#16850 was basically completely broken...

### CODE REVIEW

**Comment by luke-jr** [comment]
This fixes it, but the stupid linters don't like it:

```
test/functional/test_framework/messages.py:46: unused variable 'NODE_GETUTXO' (60% confidence)
```

Can I just get rid of the linter, or is there another approach that would be preferred?

**Comment by promag** [comment]
You could add to test/lint/lint-python-dead-code-whitelist?

**Comment by luke-jr** [comment]
There, got linter happy.

**Comment by theStack** [comment]
ACK 78487ef

That was quite an interesting chain of coincidences and fails; the fact that the tested servicesflags (0x809) didn't contain any hexadecimal symbols `a-f` covered the wrong base for the hexstring-to-integer conversion, which in turn was covered by a too weak assertion policy in `assert_net_servicesnames()`, also allowing _more_ names in the list than just the ones represented in the flags. An additional assertion `bin(servicesflags).count('1') == len(servicesnames)` would have also solved the problem, but the PR approach, getting rid of all the if statements, is more generic and flexible.

**Comment by MarcoFalke** [comment]
I wouldn't be surprised if other users also stepped into this trap. The field should probably be a normal json number, not a hex encoded string version of the int.

Concept ACK anyway

**Comment by luke-jr** [comment]
Yes, agree it should be a Number ideally.

**Comment by MarcoFalke** [comment]
unsigned ACK 78487ef8ae9d56d3285955c258c8fe0abf55b516 (looked at the diff on GitHub)

File: test/lint/lint-python-dead-code-whitelist [hunk 1]
```diff
@@ -12,6 +12,8 @@ InvalidOPIFConstruction  # unused class (test/functional/data/invalid_txs.py)
 _.is_compressed  # unused property (test/functional/test_framework/key.py)
 legacy  # unused variable (test/functional/test_framework/address.py)
 msg_generic  # unused class (test/functional/test_framework/messages.py)
+NODE_BLOOM  # actually used (test/functional/test_framework/messages.py)
+NODE_GETUTXO  # actually used (test/functional/test_framework/messages.py)
```
Comment (MarcoFalke) [inline test/lint/lint-python-dead-code-whitelist]: This file was deleted in `master`. Mind to remove those changes again and force push? (No need to rebase, if you don't want to)

**Comment by laanwj** [comment]
Closing in favor of #16991

**Pull request closed without merging**